libraries/common/components/Widgets/index.jsx

```diff
@@ -64,30 +68,94 @@ const createArrayOfElements = (widgets, components) => (
     );
   })
 );
-
 /**
  * The Widgets component.
- * @param {Object} props The component props.
- * @returns {JSX}
  */
-const Widgets = ({ widgets, components }) => {
-  if (!widgets) {
-    return null;
+class Widgets extends Component {
+  static propTypes = {
+    components: PropTypes.shape(),
+    widgets: PropTypes.arrayOf(PropTypes.shape()),
+  };
+  static defaultProps = {
+    components: null,
+    widgets: null,
+  };
+
+  /**
+   * Checks if any widget is schedulable.
+   * @param {Array} widgets Array of widgets.
+   * @returns {boolean}
+   */
+  static hasSchedulableWidgets(widgets) {
+    let result = false;
```

Comment:
maybe this method could be condensed into 
`static hasSchedulableWidgets(widgets) {
    return (widgets || []).some((w) => {
      if (w.settings.plan) {
        return true;
      }
      return false;
    });
  }`